lib/SIL/Utils/OwnershipUtils.cpp

```diff
@@ -2405,42 +2405,52 @@ bool swift::isRedundantMoveValue(MoveValueInst *mvi) {
 
   // The move doesn't alter constraints: ownership and lexicality match.
 
+  // Before checking whether escaping matches, check whether the move_value is
+  // redundant regardless on account of how its uses are limited.
+  //
+  // At this point, ownership and lexicality are known to match.  If the
+  // original value doesn't escape, then merging the two lifetimes won't make
+  // it harder to optimize the portion of the merged lifetime corresponding to
+  // the moved-to value.  If the original's only consuming use is the
+  // move_value, then the moved-from value's lifetime couldn't be shortened
```

Comment:
Sometimes you refer to the "original" value, and other times to the "moved-from" value. If you want, when you introduce the name "original" you could clarify "original (i.e. move-from)". But otherwise I think it's helpful to stay consistent.

Reply:
Fixed.